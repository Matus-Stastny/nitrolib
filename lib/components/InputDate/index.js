"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _setDate = _interopRequireDefault(require("date-fns/setDate"));

var _setMonth = _interopRequireDefault(require("date-fns/setMonth"));

var _setYear = _interopRequireDefault(require("date-fns/setYear"));

var _rtl = require("@kiwicom/orbit-components/lib/utils/rtl");

var _Theme = require("../../records/Theme");

var _Flex = _interopRequireDefault(require("../../primitives/Flex"));

var _calculateRanges2 = _interopRequireDefault(require("../../services/dates/calculateRanges"));

var _Dates = _interopRequireDefault(require("./Dates"));

var _Months = _interopRequireDefault(require("./Months"));

var _Years = _interopRequireDefault(require("./Years"));

var SelectContainer = _styledComponents.default.div.withConfig({
  displayName: "InputDate__SelectContainer",
  componentId: "sc-14pavqy-0"
})(["width:100%;margin:0 5px;&:first-child{margin-", ":0;}&:last-child{margin-", ":0;}"],
/* sc-custom "left" */
_rtl.left,
/* sc-custom "right" */
_rtl.right);

SelectContainer.defaultProps = {
  theme: _Theme.themeDefault
};

var InputDate =
/*#__PURE__*/
function (_React$PureComponent) {
  (0, _inherits2.default)(InputDate, _React$PureComponent);

  function InputDate() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, InputDate);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(InputDate)).call.apply(_getPrototypeOf2, [this].concat(args)));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "handleChangeDate", function (ev) {
      var _this$props = _this.props,
          onChange = _this$props.onChange,
          value = _this$props.value;
      onChange((0, _setDate.default)(value, Number(ev.target.value)));
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "handleChangeMonth", function (ev) {
      var _this$props2 = _this.props,
          onChange = _this$props2.onChange,
          value = _this$props2.value;
      onChange((0, _setMonth.default)(value, Number(ev.target.value)));
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "handleChangeYear", function (ev) {
      var _this$props3 = _this.props,
          onChange = _this$props3.onChange,
          value = _this$props3.value;
      onChange((0, _setYear.default)(value, Number(ev.target.value)));
    });
    return _this;
  }

  (0, _createClass2.default)(InputDate, [{
    key: "render",
    value: function render() {
      var _this$props4 = this.props,
          id = _this$props4.id,
          value = _this$props4.value,
          min = _this$props4.min,
          max = _this$props4.max,
          format = _this$props4.format;

      var _calculateRanges = (0, _calculateRanges2.default)(min, max, value),
          dates = _calculateRanges.dates,
          months = _calculateRanges.months,
          years = _calculateRanges.years;

      var parts = {
        D: React.createElement(SelectContainer, {
          key: "D"
        }, React.createElement(_Dates.default, {
          id: id,
          value: value,
          onChange: this.handleChangeDate,
          dates: dates
        })),
        M: React.createElement(SelectContainer, {
          key: "M"
        }, React.createElement(_Months.default, {
          id: id,
          value: value,
          onChange: this.handleChangeMonth,
          months: months
        })),
        Y: React.createElement(SelectContainer, {
          key: "Y"
        }, React.createElement(_Years.default, {
          id: id,
          value: value,
          onChange: this.handleChangeYear,
          years: years
        }))
      };
      return React.createElement(_Flex.default, {
        x: "space-between"
      }, format.map(function (item) {
        return parts[item];
      }));
    }
  }]);
  return InputDate;
}(React.PureComponent);

(0, _defineProperty2.default)(InputDate, "defaultProps", {
  format: ["D", "M", "Y"]
});
var _default = InputDate;
exports.default = _default;