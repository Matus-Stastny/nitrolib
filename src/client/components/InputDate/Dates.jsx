// @flow strict
import * as React from "react";
import getDate from "date-fns/getDate";

import Select from "../Select";

type Props = {|
  id: string,
  value: Date,
  onChange: (ev: SyntheticInputEvent<HTMLSelectElement>) => void,
  dates: number[],
|};

const Dates = (props: Props) => (
  <Select id={`${props.id}-date`} value={String(getDate(props.value))} onChange={props.onChange}>
    {props.dates.map(date => (
      <option key={date} value={date}>
        {date}
      </option>
    ))}
  </Select>
);

export default Dates;
