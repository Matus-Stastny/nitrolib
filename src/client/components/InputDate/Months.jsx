// @flow strict
import * as React from "react";
import getMonth from "date-fns/getMonth";

import Select from "../Select";
import Text from "../Text";

const MONTHS = [
  __("January"),
  __("February"),
  __("March"),
  __("April"),
  __("May"),
  __("June"),
  __("July"),
  __("August"),
  __("September"),
  __("October"),
  __("November"),
  __("December"),
];

type Props = {|
  id: string,
  value: Date,
  onChange: (ev: SyntheticInputEvent<HTMLSelectElement>) => void,
  months: number[],
|};

const Months = ({ id, value, onChange, months }: Props) => (
  <Select id={`${id}-month`} value={String(getMonth(value))} onChange={onChange}>
    {months.map(month => (
      <option key={month} value={month}>
        <Text t={MONTHS[month]} />
      </option>
    ))}
  </Select>
);

export default Months;
