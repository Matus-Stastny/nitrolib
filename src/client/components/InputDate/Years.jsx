// @flow strict
import * as React from "react";
import getYear from "date-fns/getYear";

import Select from "../Select";

type Props = {|
  id: string,
  value: Date,
  onChange: (ev: SyntheticInputEvent<HTMLSelectElement>) => void,
  years: number[],
|};

const Years = (props: Props) => (
  <Select id={`${props.id}-year`} value={String(getYear(props.value))} onChange={props.onChange}>
    {props.years.map(year => (
      <option key={year} value={year}>
        {year}
      </option>
    ))}
  </Select>
);

export default Years;
