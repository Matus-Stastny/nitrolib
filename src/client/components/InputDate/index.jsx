// @flow
import * as React from "react";
import styled from "styled-components";
import setDate from "date-fns/setDate";
import setMonth from "date-fns/setMonth";
import setYear from "date-fns/setYear";
import getDate from "date-fns/getDate";
import getMonth from "date-fns/getMonth";
import getYear from "date-fns/getYear";

import Flex from "client/primitives/Flex";
import Text from "../Text";
import Select from "../Select";
import calculateRanges from "./services/calculateRanges";
import minMaxCheck from "./services/minMaxCheck";

const SelectContainer = styled.div`
  width: 100%;
  margin: 0 5px;

  &:first-child {
    margin-left: 0;
  }

  &:last-child {
    margin-right: 0;
  }
`;

const MONTHS = [
  __("January"),
  __("February"),
  __("March"),
  __("April"),
  __("May"),
  __("June"),
  __("July"),
  __("August"),
  __("September"),
  __("October"),
  __("November"),
  __("December"),
];

type Props = {
  id: string,
  value: Date,
  min: Date,
  max: Date,
  onChange: Date => void,
  mmddyyyy: boolean,
};

class InputDate extends React.PureComponent<Props> {
  static defaultProps = {
    mmddyyyy: false,
  };

  componentDidMount() {
    minMaxCheck(this.props.value, this.props.min, this.props.max, this.props.onChange);
  }

  componentWillReceiveProps(nextProps: Props) {
    minMaxCheck(nextProps.value, nextProps.min, nextProps.max, nextProps.onChange);
  }

  handleChangeDay = (ev: SyntheticInputEvent<HTMLSelectElement>) => {
    const { onChange, value } = this.props;

    onChange(setDate(value, Number(ev.target.value)));
  };

  handleChangeMonth = (ev: SyntheticInputEvent<HTMLSelectElement>) => {
    const { onChange, value } = this.props;

    onChange(setMonth(value, Number(ev.target.value)));
  };

  handleChangeYear = (ev: SyntheticInputEvent<HTMLSelectElement>) => {
    const { onChange, value } = this.props;

    onChange(setYear(value, Number(ev.target.value)));
  };

  render() {
    const { id, value, min, max, mmddyyyy } = this.props;
    const { dates, months, years } = calculateRanges(min, max, value);

    return (
      <Flex x="space-between">
        {mmddyyyy ? (
          <>
            <SelectContainer>
              <Select
                id={`${id}-month`}
                value={String(getMonth(value))}
                onChange={this.handleChangeMonth}
              >
                {months.map(month => (
                  <option key={month} value={month}>
                    <Text t={MONTHS[month]} />
                  </option>
                ))}
              </Select>
            </SelectContainer>
            <SelectContainer>
              <Select
                id={`${id}-date`}
                value={String(getDate(value))}
                onChange={this.handleChangeDay}
              >
                {dates.map(date => (
                  <option key={date} value={date}>
                    {date}
                  </option>
                ))}
              </Select>
            </SelectContainer>
          </>
        ) : (
          <>
            <SelectContainer>
              <Select
                id={`${id}-date`}
                value={String(getDate(value))}
                onChange={this.handleChangeDay}
              >
                {dates.map(date => (
                  <option key={date} value={date}>
                    {date}
                  </option>
                ))}
              </Select>
            </SelectContainer>
            <SelectContainer>
              <Select
                id={`${id}-month`}
                value={String(getMonth(value))}
                onChange={this.handleChangeMonth}
              >
                {months.map(month => (
                  <option key={month} value={month}>
                    <Text t={MONTHS[month]} />
                  </option>
                ))}
              </Select>
            </SelectContainer>
          </>
        )}
        <SelectContainer>
          <Select id={`${id}-year`} value={String(getYear(value))} onChange={this.handleChangeYear}>
            {years.map(year => (
              <option key={year} value={year}>
                {year}
              </option>
            ))}
          </Select>
        </SelectContainer>
      </Flex>
    );
  }
}

export default InputDate;
