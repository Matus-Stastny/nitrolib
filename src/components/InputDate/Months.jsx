// @flow strict
import * as React from "react";
import getMonth from "date-fns/getMonth";

import Select from "../Select";
import { Consumer } from "../../services/intl/context";

const MONTHS = [
  __("January"),
  __("February"),
  __("March"),
  __("April"),
  __("May"),
  __("June"),
  __("July"),
  __("August"),
  __("September"),
  __("October"),
  __("November"),
  __("December"),
];

type Props = {|
  id: string,
  value: Date,
  onChange: (ev: SyntheticInputEvent<HTMLSelectElement>) => void,
  months: number[],
|};

const Months = ({ id, value, onChange, months }: Props) => (
  <Consumer>
    {({ translate }) => (
      <Select id={`${id}-month`} value={String(getMonth(value))} onChange={onChange}>
        {months.map(month => (
          <option key={month} value={month}>
            {translate(MONTHS[month])}
          </option>
        ))}
      </Select>
    )}
  </Consumer>
);

export default Months;
