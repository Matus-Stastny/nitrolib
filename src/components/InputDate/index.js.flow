// @flow strict
import * as React from "react";

type Props = {|
  id: string,
  value: Date,
  min: Date,
  max: Date,
  onChange: Date => void,
  mmddyyyy: boolean,
|};

declare export default React.ComponentType<Props>;
