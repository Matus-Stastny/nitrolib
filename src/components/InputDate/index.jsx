// @flow strict
import * as React from "react";
import styled from "styled-components";
import setDate from "date-fns/setDate";
import setMonth from "date-fns/setMonth";
import setYear from "date-fns/setYear";

import * as rtl from "../../styles/rtl";
import { themeDefault } from "../../records/Theme";
import Flex from "../../primitives/Flex";
import calculateRanges from "./services/calculateRanges";
import Dates from "./Dates";
import Months from "./Months";
import Years from "./Years";

const SelectContainer = styled.div`
  width: 100%;
  margin: 0 5px;

  &:first-child {
    margin-${rtl.left}: 0;
  }

  &:last-child {
    margin-${rtl.right}: 0;
  }
`;

SelectContainer.defaultProps = {
  theme: themeDefault,
};

type Props = {|
  id: string,
  value: Date,
  min: Date,
  max: Date,
  onChange: Date => void,
  mmddyyyy: boolean,
|};

class InputDate extends React.PureComponent<Props> {
  static defaultProps = {
    mmddyyyy: false,
  };

  handleChangeDate = (ev: SyntheticInputEvent<HTMLSelectElement>) => {
    const { onChange, value } = this.props;

    onChange(setDate(value, Number(ev.target.value)));
  };

  handleChangeMonth = (ev: SyntheticInputEvent<HTMLSelectElement>) => {
    const { onChange, value } = this.props;

    onChange(setMonth(value, Number(ev.target.value)));
  };

  handleChangeYear = (ev: SyntheticInputEvent<HTMLSelectElement>) => {
    const { onChange, value } = this.props;

    onChange(setYear(value, Number(ev.target.value)));
  };

  render() {
    const { id, value, min, max, mmddyyyy } = this.props;
    const { dates, months, years } = calculateRanges(min, max, value);

    return (
      <Flex x="space-between">
        {mmddyyyy ? (
          <>
            <SelectContainer>
              <Months id={id} value={value} onChange={this.handleChangeMonth} months={months} />
            </SelectContainer>
            <SelectContainer>
              <Dates id={id} value={value} onChange={this.handleChangeDate} dates={dates} />
            </SelectContainer>
          </>
        ) : (
          <>
            <SelectContainer>
              <Dates id={id} value={value} onChange={this.handleChangeDate} dates={dates} />
            </SelectContainer>
            <SelectContainer>
              <Months id={id} value={value} onChange={this.handleChangeMonth} months={months} />
            </SelectContainer>
          </>
        )}
        <SelectContainer>
          <Years id={id} value={value} onChange={this.handleChangeYear} years={years} />
        </SelectContainer>
      </Flex>
    );
  }
}

export default InputDate;
