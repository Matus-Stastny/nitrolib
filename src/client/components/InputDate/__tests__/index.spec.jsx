// @flow strict
import * as React from "react";
import { shallow } from "enzyme";
import addYears from "date-fns/addYears";
import setDate from "date-fns/setDate";
import setMonth from "date-fns/setMonth";
import setYear from "date-fns/setYear";

import InputDate from "../index";

const NOW = new Date(Date.UTC(2018, 1, 1));

describe("#InputDate", () => {
  test("render regular", () => {
    const wrapper = shallow(
      <InputDate
        id="test"
        value={NOW}
        onChange={jest.fn()}
        min={addYears(NOW, -1)}
        max={addYears(NOW, 2)}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  test("render mmddyyyy", () => {
    const wrapper = shallow(
      <InputDate
        id="test"
        value={NOW}
        onChange={jest.fn()}
        min={addYears(NOW, -1)}
        max={addYears(NOW, 2)}
        mmddyyyy
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  test("check min on mount", () => {
    const min = addYears(NOW, -1);
    const onChange = jest.fn();
    shallow(
      <InputDate
        id="test"
        value={addYears(NOW, -2)}
        onChange={onChange}
        min={min}
        max={addYears(NOW, 2)}
      />,
    );

    expect(onChange).toBeCalledWith(min);
  });

  test("check max on mount", () => {
    const max = addYears(NOW, 2);
    const onChange = jest.fn();
    shallow(
      <InputDate
        id="test"
        value={addYears(NOW, 3)}
        onChange={onChange}
        min={addYears(NOW, -1)}
        max={max}
      />,
    );

    expect(onChange).toBeCalledWith(max);
  });

  test("check min on update", () => {
    const min = addYears(NOW, -1);
    const onChange = jest.fn();
    const wrapper = shallow(
      <InputDate id="test" value={NOW} onChange={onChange} min={min} max={addYears(NOW, 2)} />,
    );

    expect(onChange).not.toBeCalled();

    wrapper.setProps({ value: addYears(NOW, -2) });

    expect(onChange).toBeCalledWith(min);
  });

  test("check max on update", () => {
    const max = addYears(NOW, 2);
    const onChange = jest.fn();
    const wrapper = shallow(
      <InputDate id="test" value={NOW} onChange={onChange} min={addYears(NOW, -1)} max={max} />,
    );

    expect(onChange).not.toBeCalled();

    wrapper.setProps({ value: addYears(NOW, 3) });

    expect(onChange).toBeCalledWith(max);
  });

  test("handle change date", () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <InputDate
        id="test"
        value={NOW}
        onChange={onChange}
        min={addYears(NOW, -1)}
        max={addYears(NOW, 2)}
      />,
    );

    wrapper.find("#test-date").simulate("change", { target: { value: "10" } });

    expect(onChange).toBeCalledWith(setDate(NOW, 10));
  });

  test("handle change month", () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <InputDate
        id="test"
        value={NOW}
        onChange={onChange}
        min={addYears(NOW, -1)}
        max={addYears(NOW, 2)}
      />,
    );

    wrapper.find("#test-month").simulate("change", { target: { value: "10" } });

    expect(onChange).toBeCalledWith(setMonth(NOW, 10));
  });

  test("handle change year", () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <InputDate
        id="test"
        value={NOW}
        onChange={onChange}
        min={addYears(NOW, -1)}
        max={addYears(NOW, 2)}
      />,
    );

    wrapper.find("#test-year").simulate("change", { target: { value: "2020" } });

    expect(onChange).toBeCalledWith(setYear(NOW, 2020));
  });
});
