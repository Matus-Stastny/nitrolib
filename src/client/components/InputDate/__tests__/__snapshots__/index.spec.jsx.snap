// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#InputDate render mmddyyyy 1`] = `
<Flex
  direction="row"
  x="space-between"
  y="flex-start"
>
  <React.Fragment>
    <InputDate__SelectContainer>
      <Select
        id="test-month"
        onChange={[Function]}
        value="1"
      >
        <option
          key="0"
          value={0}
        >
          <Text
            html={false}
            t="January"
            values={Object {}}
          />
        </option>
        <option
          key="1"
          value={1}
        >
          <Text
            html={false}
            t="February"
            values={Object {}}
          />
        </option>
        <option
          key="2"
          value={2}
        >
          <Text
            html={false}
            t="March"
            values={Object {}}
          />
        </option>
        <option
          key="3"
          value={3}
        >
          <Text
            html={false}
            t="April"
            values={Object {}}
          />
        </option>
        <option
          key="4"
          value={4}
        >
          <Text
            html={false}
            t="May"
            values={Object {}}
          />
        </option>
        <option
          key="5"
          value={5}
        >
          <Text
            html={false}
            t="June"
            values={Object {}}
          />
        </option>
        <option
          key="6"
          value={6}
        >
          <Text
            html={false}
            t="July"
            values={Object {}}
          />
        </option>
        <option
          key="7"
          value={7}
        >
          <Text
            html={false}
            t="August"
            values={Object {}}
          />
        </option>
        <option
          key="8"
          value={8}
        >
          <Text
            html={false}
            t="September"
            values={Object {}}
          />
        </option>
        <option
          key="9"
          value={9}
        >
          <Text
            html={false}
            t="October"
            values={Object {}}
          />
        </option>
        <option
          key="10"
          value={10}
        >
          <Text
            html={false}
            t="November"
            values={Object {}}
          />
        </option>
        <option
          key="11"
          value={11}
        >
          <Text
            html={false}
            t="December"
            values={Object {}}
          />
        </option>
      </Select>
    </InputDate__SelectContainer>
    <InputDate__SelectContainer>
      <Select
        id="test-date"
        onChange={[Function]}
        value="1"
      >
        <option
          key="1"
          value={1}
        >
          1
        </option>
        <option
          key="2"
          value={2}
        >
          2
        </option>
        <option
          key="3"
          value={3}
        >
          3
        </option>
        <option
          key="4"
          value={4}
        >
          4
        </option>
        <option
          key="5"
          value={5}
        >
          5
        </option>
        <option
          key="6"
          value={6}
        >
          6
        </option>
        <option
          key="7"
          value={7}
        >
          7
        </option>
        <option
          key="8"
          value={8}
        >
          8
        </option>
        <option
          key="9"
          value={9}
        >
          9
        </option>
        <option
          key="10"
          value={10}
        >
          10
        </option>
        <option
          key="11"
          value={11}
        >
          11
        </option>
        <option
          key="12"
          value={12}
        >
          12
        </option>
        <option
          key="13"
          value={13}
        >
          13
        </option>
        <option
          key="14"
          value={14}
        >
          14
        </option>
        <option
          key="15"
          value={15}
        >
          15
        </option>
        <option
          key="16"
          value={16}
        >
          16
        </option>
        <option
          key="17"
          value={17}
        >
          17
        </option>
        <option
          key="18"
          value={18}
        >
          18
        </option>
        <option
          key="19"
          value={19}
        >
          19
        </option>
        <option
          key="20"
          value={20}
        >
          20
        </option>
        <option
          key="21"
          value={21}
        >
          21
        </option>
        <option
          key="22"
          value={22}
        >
          22
        </option>
        <option
          key="23"
          value={23}
        >
          23
        </option>
        <option
          key="24"
          value={24}
        >
          24
        </option>
        <option
          key="25"
          value={25}
        >
          25
        </option>
        <option
          key="26"
          value={26}
        >
          26
        </option>
        <option
          key="27"
          value={27}
        >
          27
        </option>
        <option
          key="28"
          value={28}
        >
          28
        </option>
      </Select>
    </InputDate__SelectContainer>
  </React.Fragment>
  <InputDate__SelectContainer>
    <Select
      id="test-year"
      onChange={[Function]}
      value="2018"
    >
      <option
        key="2017"
        value={2017}
      >
        2017
      </option>
      <option
        key="2018"
        value={2018}
      >
        2018
      </option>
      <option
        key="2019"
        value={2019}
      >
        2019
      </option>
      <option
        key="2020"
        value={2020}
      >
        2020
      </option>
    </Select>
  </InputDate__SelectContainer>
</Flex>
`;

exports[`#InputDate render regular 1`] = `
<Flex
  direction="row"
  x="space-between"
  y="flex-start"
>
  <React.Fragment>
    <InputDate__SelectContainer>
      <Select
        id="test-date"
        onChange={[Function]}
        value="1"
      >
        <option
          key="1"
          value={1}
        >
          1
        </option>
        <option
          key="2"
          value={2}
        >
          2
        </option>
        <option
          key="3"
          value={3}
        >
          3
        </option>
        <option
          key="4"
          value={4}
        >
          4
        </option>
        <option
          key="5"
          value={5}
        >
          5
        </option>
        <option
          key="6"
          value={6}
        >
          6
        </option>
        <option
          key="7"
          value={7}
        >
          7
        </option>
        <option
          key="8"
          value={8}
        >
          8
        </option>
        <option
          key="9"
          value={9}
        >
          9
        </option>
        <option
          key="10"
          value={10}
        >
          10
        </option>
        <option
          key="11"
          value={11}
        >
          11
        </option>
        <option
          key="12"
          value={12}
        >
          12
        </option>
        <option
          key="13"
          value={13}
        >
          13
        </option>
        <option
          key="14"
          value={14}
        >
          14
        </option>
        <option
          key="15"
          value={15}
        >
          15
        </option>
        <option
          key="16"
          value={16}
        >
          16
        </option>
        <option
          key="17"
          value={17}
        >
          17
        </option>
        <option
          key="18"
          value={18}
        >
          18
        </option>
        <option
          key="19"
          value={19}
        >
          19
        </option>
        <option
          key="20"
          value={20}
        >
          20
        </option>
        <option
          key="21"
          value={21}
        >
          21
        </option>
        <option
          key="22"
          value={22}
        >
          22
        </option>
        <option
          key="23"
          value={23}
        >
          23
        </option>
        <option
          key="24"
          value={24}
        >
          24
        </option>
        <option
          key="25"
          value={25}
        >
          25
        </option>
        <option
          key="26"
          value={26}
        >
          26
        </option>
        <option
          key="27"
          value={27}
        >
          27
        </option>
        <option
          key="28"
          value={28}
        >
          28
        </option>
      </Select>
    </InputDate__SelectContainer>
    <InputDate__SelectContainer>
      <Select
        id="test-month"
        onChange={[Function]}
        value="1"
      >
        <option
          key="0"
          value={0}
        >
          <Text
            html={false}
            t="January"
            values={Object {}}
          />
        </option>
        <option
          key="1"
          value={1}
        >
          <Text
            html={false}
            t="February"
            values={Object {}}
          />
        </option>
        <option
          key="2"
          value={2}
        >
          <Text
            html={false}
            t="March"
            values={Object {}}
          />
        </option>
        <option
          key="3"
          value={3}
        >
          <Text
            html={false}
            t="April"
            values={Object {}}
          />
        </option>
        <option
          key="4"
          value={4}
        >
          <Text
            html={false}
            t="May"
            values={Object {}}
          />
        </option>
        <option
          key="5"
          value={5}
        >
          <Text
            html={false}
            t="June"
            values={Object {}}
          />
        </option>
        <option
          key="6"
          value={6}
        >
          <Text
            html={false}
            t="July"
            values={Object {}}
          />
        </option>
        <option
          key="7"
          value={7}
        >
          <Text
            html={false}
            t="August"
            values={Object {}}
          />
        </option>
        <option
          key="8"
          value={8}
        >
          <Text
            html={false}
            t="September"
            values={Object {}}
          />
        </option>
        <option
          key="9"
          value={9}
        >
          <Text
            html={false}
            t="October"
            values={Object {}}
          />
        </option>
        <option
          key="10"
          value={10}
        >
          <Text
            html={false}
            t="November"
            values={Object {}}
          />
        </option>
        <option
          key="11"
          value={11}
        >
          <Text
            html={false}
            t="December"
            values={Object {}}
          />
        </option>
      </Select>
    </InputDate__SelectContainer>
  </React.Fragment>
  <InputDate__SelectContainer>
    <Select
      id="test-year"
      onChange={[Function]}
      value="2018"
    >
      <option
        key="2017"
        value={2017}
      >
        2017
      </option>
      <option
        key="2018"
        value={2018}
      >
        2018
      </option>
      <option
        key="2019"
        value={2019}
      >
        2019
      </option>
      <option
        key="2020"
        value={2020}
      >
        2020
      </option>
    </Select>
  </InputDate__SelectContainer>
</Flex>
`;
